substrate/frame/data-market/src/lib.rs

```diff
@@ -28,266 +35,572 @@ pub mod pallet {
 
-		/// The total fee ratio, defined as 100% and represented by the value 10,000.
-		#[pallet::constant]
-		type TotalFeeRatio: Get<u32>;
+		/// The type used to tokenize the asset balance.
+		type Assets: Inspect<Self::AccountId> + Mutate<Self::AccountId>;
 
-		/// The minimum platform fee ratio, set at a fixed 10%.
+		// The maximum quantity of data that can be purchased
 		#[pallet::constant]
-		type MinPlatformFeeRatio: Get<u32>;
+		type MaxPurchaseQuantity: Get<Quantity>;
 	}
 
+	// The Next value of contract id
+	#[pallet::storage]
+	#[pallet::getter(fn get_next_contract_id)]
+	pub(super) type NextContractId<T: Config> = StorageValue<_, ContractId, ValueQuery>;
+
+	// The Data Trade Records
+	#[pallet::storage]
+	#[pallet::getter(fn get_data_trade_records)]
+	pub type DataTradeRecords<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, ContractId, Twox64Concat, T::AccountId, ()>;
+
+	// The Trade Count for Contract
 	#[pallet::storage]
-	pub(super) type NextPurchaseId<T: Config> = StorageValue<_, PurchaseId, ValueQuery>;
+	#[pallet::getter(fn get_trade_count_for_purchase)]
+	pub type TradeCountForContract<T: Config> =
+		StorageMap<_, Twox64Concat, ContractId, Quantity, ValueQuery>;
 
+	// The Data Delegate Contracts
 	#[pallet::storage]
-	#[pallet::getter(fn data_purchase_registers)]
-	pub type DataPurchaseRegisters<T: Config> = StorageMap<
+	#[pallet::getter(fn get_data_delegate_contracts)]
+	pub(super) type DataDelegateContracts<T: Config> = StorageMap<
 		_,
 		Twox64Concat,
-		PurchaseId,
-		DataPurchaseRegisterDetails<
-			T::AccountId,
-			BlockNumberFor<T>,
-			<T as pallet_assets::Config>::Balance,
-			AnyText,
-		>,
+		ContractId,
+		DataDelegateContractDetail<T::AccountId, BlockNumberFor<T>>,
 		OptionQuery,
 	>;
 
+	// The Data Purchase Contracts
 	#[pallet::storage]
-	#[pallet::getter(fn data_trade_records)]
-	pub type DataTradeRecords<T: Config> =
-		StorageDoubleMap<_, Twox64Concat, PurchaseId, Twox64Concat, T::AccountId, ()>;
+	#[pallet::getter(fn get_data_purchase_contracts)]
+	pub(super) type DataPurchaseContracts<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractId,
+		DataPurchaseContractDetail<T::AccountId, BlockNumberFor<T>, AssetBalanceOf<T>>,
+		OptionQuery,
+	>;
+
+	// The Data Delegate Contract List
+	#[pallet::storage]
+	#[pallet::getter(fn get_data_delegate_contract_list)]
+	pub(super) type DataDelegateContractList<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, Vec<ContractId>, ValueQuery>;
+
+	// The Data Purchase Contract List
+	#[pallet::storage]
+	#[pallet::getter(fn get_data_purchase_contract_list)]
+	pub(super) type DataPurchaseContractList<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, Vec<ContractId>, ValueQuery>;
+
+	// The Contract Status
+	#[pallet::storage]
+	#[pallet::getter(fn get_contract_status)]
+	pub(super) type ContractStatus<T: Config> =
+		StorageMap<_, Twox64Concat, ContractId, ContractSigner<T>, OptionQuery>;
```

Comment:
차라리 해당 `ContractId` 에 해당하는 `ContractDetail` 구조체에 `signed_status` 필드를 추가해서 메소드를 추가 구현하는게 낫지 않을까요

```suggestion
	ContractDetail {
        ..
        signed_status: [(owner, true/false), (agency, true/false)]
    }
```